wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java

```diff
@@ -534,14 +534,17 @@ public Response importZip(String wsPath, InputStream zip, Boolean skipFirstLevel
 
     fsManager.unzip(wsPath, zip, skipFirstLevel);
 
-    return Response.created(
-            serviceContext
-                .getBaseUriBuilder()
-                .clone()
-                .path(ProjectService.class)
-                .path(ProjectService.class, "getChildren")
-                .build(new String[] {wsPath.substring(1)}, false))
-        .build();
+    Response response =
+        Response.created(
+                serviceContext
+                    .getBaseUriBuilder()
+                    .clone()
+                    .path(ProjectService.class)
+                    .path(ProjectService.class, "getChildren")
+                    .build(new String[] {wsPath.substring(1)}, false))
+            .build();
+    eventService.publish(new ProjectCreatedEvent(wsPath));
```

Comment:
i am not sure if it is a good place to publish `ProjectCreatedEvent`  @vparfonov WDYT ?

Reply:
I am ok for firing event from this class we already do it in several places, but i think we need to fire it only if we create project in ROOT

Reply:
@vparfonov I don't really see the logic of that, but anyways...we need to have this work to continue fixing selenium tests. We can clear up the difference once we have a fix for https://github.com/eclipse/che/issues/9924